iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Branch.java

```diff
@@ -164,53 +169,63 @@ default ApparentPowerLimits getApparentPowerLimits1() {
 
-    default ActivePowerLimits getActivePowerLimits2() {
-        return null;
-    }
+    CurrentLimits getNullableCurrentLimits2();
 
-    default ApparentPowerLimits getApparentPowerLimits2() {
-        return null;
-    }
+    Optional<ActivePowerLimits> getActivePowerLimits2();
+
+    ActivePowerLimits getNullableActivePowerLimits2();
+
+    Optional<ApparentPowerLimits> getApparentPowerLimits2();
+
+    ApparentPowerLimits getNullableApparentPowerLimits2();
 
     CurrentLimitsAdder newCurrentLimits2();
 
     ActivePowerLimitsAdder newActivePowerLimits2();
 
     ApparentPowerLimitsAdder newApparentPowerLimits2();
 
-    default CurrentLimits getCurrentLimits(Branch.Side side) {
-        if (side == Branch.Side.ONE) {
-            return getCurrentLimits1();
-        } else if (side == Branch.Side.TWO) {
-            return getCurrentLimits2();
+    default Optional<CurrentLimits> getCurrentLimits(Branch.Side side) {
+        switch (side) {
+            case ONE:
+                return getCurrentLimits1();
+            case TWO:
+                return getCurrentLimits2();
+            default:
+                throw new UnsupportedOperationException(String.format("Side %s not supported", side.name()));
         }
-        throw new AssertionError("Unexpected side: " + side);
     }
 
-    default ActivePowerLimits getActivePowerLimits(Branch.Side side) {
-        if (side == Branch.Side.ONE) {
-            return getActivePowerLimits1();
-        } else if (side == Branch.Side.TWO) {
-            return getActivePowerLimits2();
+    default Optional<ActivePowerLimits> getActivePowerLimits(Branch.Side side) {
+        switch (side) {
+            case ONE:
+                return getActivePowerLimits1();
+            case TWO:
+                return getActivePowerLimits2();
+            default:
+                throw new UnsupportedOperationException(String.format("Side %s not supported", side.name()));
         }
-        throw new AssertionError("Unexpected side: " + side);
     }
 
-    default ApparentPowerLimits getApparentPowerLimits(Branch.Side side) {
-        if (side == Branch.Side.ONE) {
-            return getApparentPowerLimits1();
-        } else if (side == Branch.Side.TWO) {
-            return getApparentPowerLimits2();
+    default Optional<ApparentPowerLimits> getApparentPowerLimits(Branch.Side side) {
+        switch (side) {
+            case ONE:
+                return getApparentPowerLimits1();
+            case TWO:
+                return getApparentPowerLimits2();
+            default:
+                throw new UnsupportedOperationException(String.format("Side %s not supported", side.name()));
         }
-        throw new AssertionError("Unexpected side: " + side);
     }
 
-    default LoadingLimits getLimits(LimitType type, Branch.Side side) {
+    default Optional<? extends LoadingLimits> getLimits(LimitType type, Branch.Side side) {
         switch (type) {
```

Comment:
we could use `Optional.ofNullable(getNullableLimits(...))`, or maybe you have in mind to deprecate the getNullables one day?

Reply:
I didn't want `getNullable...` methods to be called in the interface as they should not have to be implemented (they only exist for backward compatibility on groovy)